infoschema/tables.go

```diff
@@ -684,13 +684,27 @@ var filesCols = []columnInfo{
 }
 
 func dataForSchemata(schemas []*model.DBInfo) [][]types.Datum {
+
 	var rows [][]types.Datum
+
 	for _, schema := range schemas {
+
+		charset := mysql.DefaultCharset
+		collation := mysql.DefaultCollationName
+
+		if len(schema.Charset) > 0 {
+			charset = schema.Charset // Overwrite default
+		}
+
+		if len(schema.Collate) > 0 {
+			collation = schema.Collate // Overwrite default
+		}
+
 		record := types.MakeDatums(
-			catalogVal,                 // CATALOG_NAME
-			schema.Name.O,              // SCHEMA_NAME
-			mysql.DefaultCharset,       // DEFAULT_CHARACTER_SET_NAME
-			mysql.DefaultCollationName, // DEFAULT_COLLATION_NAME
+			catalogVal,    // CATALOG_NAME
+			schema.Name.O, // SCHEMA_NAME
+			charset,       // DEFAULT_CHARACTER_SET_NAME
```

Comment:
It is better to update the comment to remove the `DEFAULT` prefix.

Reply:
As I understand, the comment refers to the structure of `is.schemata`.  It remains correct with this patch:
```
mysql> SHOW CREATE TABLE schemata\G
*************************** 1. row ***************************
       Table: SCHEMATA
Create Table: CREATE TABLE `SCHEMATA` (
  `CATALOG_NAME` varchar(512) DEFAULT NULL,
  `SCHEMA_NAME` varchar(64) DEFAULT NULL,
  `DEFAULT_CHARACTER_SET_NAME` varchar(64) DEFAULT NULL,
  `DEFAULT_COLLATION_NAME` varchar(32) DEFAULT NULL,
  `SQL_PATH` varchar(512) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin
1 row in set (0.01 sec)
```